application/classes/infosec/login_service.py

```diff
@@ -17,27 +17,11 @@ def set_username(self, username):
         self.__username = username
 
     def set_password(self, password):
-        """Encrypts password and sets it"""
-        encrypted_password = self.__encryption_service.encrypt(password)
-        self.__password = encrypted_password
+        self.__password = password
     
     def get_loggedin_username(self):
         """Returns username of logged in user"""
         return self.__username
-    
-    def get_password(self):
-        """Decrypts password and returns it"""
-        decrypted_password = self.__encryption_service.decrypt(self.__password)
-        return decrypted_password
-    
-    def check_password(self, password):
```

Comment:
I think we need this method (check_password()), as the commandline_interface calls on it when a user requests to change their phrase or password to authenticate the user.

Reply:
Unless we implement that in a different manner, I think we'd need the "get_password()" method above as well. Let me know what you think.

Reply:
I'll add those back in, so we can use them. But one thing I noticed in this version was that the password was being encrypted by the bcrypt library and the encryption service, but in the authenticate credentials method, only bcrypt was being used to decrypt the password, so i think we might need to add the encyption decrypt to the getter. The check_password method wasn't being called anywjere. I have a feeling this is a slighly older version of the file though.

Reply:
Copy that! I think you're right; the problem might be the self.__password value being accessed directly in the "authenticate_user_credentials()" method, which means it's still encrypted by the encryption service. Using the getter with the `decrypted_password = self.__encryption_service.decrypt(self.__password)` line would probably solve the problem.

Reply:
The `check_password()` method is being called in the `change_password()` method in the CLI class, and I believe, in the `change_phrase()` method of the CLI class as well.

![Screenshot 2023-07-10 100945](https://github.com/michaelsammueller/secure-software-development/assets/34138597/0121963c-0b04-46c7-af6d-b1e7ca3e92bc)

I'll play around with it once I'm home; sorry for the inconvenience!

Reply:
![Screenshot_20230710-082553_Chrome](https://github.com/michaelsammueller/secure-software-development/assets/81972602/a10739cd-0cb1-41af-a898-3ec23deac2a1)
No worries, it did confuse me, so I might have misunderstood

Reply:
My naming conventions probably aren't ideal because there's a `change_password()` method in the `login_service.py` and a `change_password()` method in the `commandline_interface.py` file. The former doesn't call on the `check_password()` method, but the latter does. 🤯😂